models/error.go

```diff
@@ -1371,6 +1372,48 @@ func (err ErrMergePushOutOfDate) Error() string {
 	return fmt.Sprintf("Merge PushOutOfDate Error: %v: %s\n%s", err.Err, err.StdErr, err.StdOut)
 }
 
+// ErrPushRejected represents an error if merging fails due to unrelated histories
+type ErrPushRejected struct {
+	Style  MergeStyle
+	StdOut string
+	StdErr string
+	Err    error
+}
+
+// IsErrPushRejected checks if an error is a ErrPushRejected.
+func IsErrPushRejected(err error) bool {
+	_, ok := err.(ErrPushRejected)
+	return ok
+}
+
+func (err ErrPushRejected) Error() string {
+	return fmt.Sprintf("Merge PushRejected Error: %v: %s\n%s", err.Err, err.StdErr, err.StdOut)
+}
+
+// Message returns the cleaned message from Error
+func (err ErrPushRejected) Message() string {
+	messageBuilder := &strings.Builder{}
+	i := 0
+	for {
+		if len(err.StdErr) <= i+8 {
+			break
+		}
+		if err.StdErr[i:i+8] != "remote: " {
+			break
+		}
+		i = i + 8
+		nl := strings.IndexByte(err.StdErr[i:], '\n')
```

Comment:
I'm assuming there's no chance for EOL style problems here. 😅

Reply:
I think git handles that for us.